robottelo/hosts.py

```diff
@@ -135,13 +136,20 @@ def __init__(self, hostname, auth=None, **kwargs):
         elif isinstance(auth, str):
             # key file based authentication
             kwargs.update({'key_filename': auth})
+        self._satellite = kwargs.get('satellite')
         super().__init__(hostname=hostname, **kwargs)
 
+    @property
+    def satellite(self):
+        if not self._satellite:
+            self._satellite = Satellite()
```

Comment:
if you dont provide the satellite hostname argument here, this will always point to the `default_sat` / `setting.server.hostname` which is not being set by satellite_factory(in case you of destructive sat).

Ofcourse you want it to point to destructive_sat since the host will belong to destructive_sat that time not to default_sat.

Reply:
I don't think the property should resolve this automatically. If a sat instance wasn't passed on instantiation, this function should return `None`. Having a setter would be appropriate as well, so a sat can be set on the host after instantiation.

Not all host instances will necessarily have a satellite that they're registered against. With this behavior, its hard/impossible to tell.